examples/feature_engineering.py

```diff
@@ -0,0 +1,303 @@
+        - table_name (str): name of the data table to be retrieved
+        - nrows (int, optional): defaults to all rows; otherwise, number of rows to load
+        - exclude_cols (List[str], optional): columns to be excluded, by name
+
+    Returns:
+        - pd.DataFrame: the contents of the CSV
+    """
+    filepath = os.path.join("data", rootdir, "core", table_name + ".csv")
+    df = pd.read_csv(filepath, nrows=None)
+    return df.drop(exclude_cols, axis=1)
+
+
+@task
+def get_colnames(table: pd.DataFrame) -> List[str]:
+    """
+    Get column names from a DataFrame.
+    """
+    return list(table.columns)
+
+
+@task
+def get_cols(table: pd.DataFrame) -> List[np.ndarray]:
+    """
+    Get list of arrays corresponding to the columns of a DataFrame.
+    """
+    return list(table.values.T)
+
+
+# data manipulation tasks
+@task
+def impute(data: np.ndarray, replacement_dict: Dict[Any, Any]) -> np.ndarray:
+    """
+    Replace any instances of a set of values with corresponding new values.
+
+    The replacements are provided in a dictionary.
+    The keys are the values to be replaced,
+    and the values are the values to replace with.
+
+    Args:
+        - data (np.ndarray): data in which the replacements should be made
+        - replacement_dict (Dict[Any, Any]): what to replace, and what to replace it with
+
+    Returns:
+        - np.ndarray: data with values replaced
+    """
+    for old, new in replacement_dict.items():
+        if np.isnan(old):
+            data[pd.isnull(data)] = new
+        else:
+            data[data == old] = new
+    return data
+
+
+@task
+def cast(data: np.ndarray, new_type: type) -> np.ndarray:
+    """
+    Cast all the values of an array to a new data type.
+
+    Args:
+        - data (np.ndarray): data to be cast
+        - new_type (type): type to be cast to
+
+    Returns:
+        - np.ndarray: the data in its new type
+    """
+    return data.astype(new_type)
+
+
+@task
+def get(values: List[np.ndarray], keys: List[str], key: str) -> np.ndarray:
+    """
+    Retrieve the value associated with a key given a list of keys and list of values.
+
+    Essentially a utility to make use of dicts that have been broken into
+    a list of keys and a list of values.
+
+    Args:
+        - values (List[np.ndarray]): values associated with keys
+        - keys (List[str]): keys to choose from
+        - key (str): the chosen key
+
+    Returns:
+        - np.ndarray: the value associated with the chosen key
+    """
+    return values[keys.index(key)]
+
+
+@task
+def concatenate(arrays: List[np.ndarray]) -> np.ndarray:
+    """
+    Concatenate multiple arrays along the last axis.
+    for 1D arrays, make a 2D array of columns.
+    """
+    for i, a in enumerate(arrays):
+        if len(a.shape) == 1:
+            arrays[i] = np.expand_dims(a, -1)
+    return np.concatenate(arrays, axis=-1)
+
+
+@task
+def dot(A: np.ndarray, B: np.ndarray) -> np.ndarray:
```

Comment:
"Small" tasks like this feel trivial. Depending on how explicit you want to be, it would be nice to replace this with:

```python
prefect.task(np.dot)(a, b)
```

However you would lose a little of your own documentation and typing. 

Unfortunately, in this specific case, it looks like `np.dot` is an "unsupported callable" for our signature introspection, so while this would work for Python functions it doesn't work here -- cc @cicdw, hopefully we can work around this.

Reply:
Another side note on that is that I originally had `dot` making use of `np.linalg.multi_dot` so it could take an arbitrary number of arrays. I switched to this signature because I was mistakenly thinking that not having *args was preventing me from passing a variable number of arrays (I should probably avoid cranking these out in one sitting in the future haha). I'm sure `multi_dot` is another unsupported callable, but I might just make that change quickly now that I've seen it.

Reply:
Yea, @jlowin maybe we can open an issue to research this (handling "unspported callables").  I believe this occurs with functions which are connected directly to C, and consequently `inspect` can't determine their call signature.  We might lose some helpful error messages if a user provides an incorrect kwarg to the functions if we decide to support that.